crates/burn-jit/src/kernel/conv/conv2d/implicit_gemm.rs

```diff
@@ -0,0 +1,343 @@
+use burn_tensor::{
+    ops::{conv::calculate_conv_output_size, ConvOptions, FloatTensorOps},
+    Shape,
+};
+use cmma::{Matrix, MatrixIdent, MatrixLayout};
+use cubecl::{
+    comptime, cube,
+    ir::{Elem, FloatKind},
+    prelude::*,
+    Compiler, CubeCount, CubeDim, Feature,
+};
+use half::f16;
+
+use crate::{
+    ops::{numeric::empty_device, permute, reshape},
+    tensor::JitTensor,
+    FloatElement, IntElement, JitBackend, JitRuntime,
+};
+
+const WARP_SIZE: u32 = 32;
+
+const CMMA_M: u32 = 16;
+const CMMA_N: u32 = 16;
+const CMMA_K: u32 = 16;
+const CMMA_INPUT_TILE_SIZE: u32 = CMMA_M * CMMA_K;
+const CMMA_FILTER_TILE_SIZE: u32 = CMMA_K * CMMA_N;
+const CMMA_OUT_TILE_SIZE: u32 = CMMA_M * CMMA_N;
+
+const WARPS_PER_CUBE: u32 = 8;
+
+const CUBE_DIM_X: u32 = 128;
```

Comment:
I'm genuinely curious, why did you choose (128, 2) and not (32, 8) for instance, is it better?

Reply:
It makes sense when you look at the dimensionality of the matrix. The `x` dimension is `batch_size * out_h * out_w` which is much larger than the `y` dimension, which is `in_channels * kernel_h * kernel_w`. `128 x 2` is a decent approximation of the relative sizes between `M` and `N`. This lopsidedness caused issues before with the `im2col` because the matmul kernel would pick something like `144k x 1` as its cube count because it uses a square `CubeDim` but the actual matrix is anything but square.